### PR TITLE
Static mocking documentation updates

### PR DESCRIPTION
**Pull request by AndreasTu**
Moved static mocking into own section.
Added sample to do stubbing of all static methods.

### CODE REVIEW

**Comment by AndreasTu** [comment]
This shall fix the issues mentioned in https://github.com/spockframework/spock/pull/1756#issuecomment-1974259165

File: docs/interaction_based_testing.adoc [hunk 1]
```diff
@@ -1027,70 +1100,8 @@ new RealSubscriber("Fred") >> new RealSubscriber("Barney")
...
-[source,groovy,indent=0]
-----
-include::{sourcedir}/interaction/StaticSpyDocSpec.groovy[tag=mock-static-different-thread]
-----
-
-The interface `spock.mock.IStaticMock` provides API to activate a static mock on other threads.
-
-==== Global Groovy Mocks for Static Methods
+[[global-groovy-mock-static-methods]]
+=== Global Groovy Mocks for Static Methods
```
Comment (leonard84) [inline docs/interaction_based_testing.adoc:1104]: Should we add a link to the other section?

**Review by leonard84 — APPROVED** [review]

File: docs/interaction_based_testing.adoc (hunk 1 above)
Comment (AndreasTu) [reply docs/interaction_based_testing.adoc:1104]: I have added a link.

**Review by leonard84 — APPROVED** [review]

**Comment by Vampire** [comment]
Was it intentional, that you lost the note about parallel test execution with Groovy Global Mocks? :-)

**Comment by leonard84** [comment]
I guess not, good catch.

**Comment by AndreasTu** [comment]
Yes it was, because it is already mentioned in the section `Global mocks and parallel execution`, and we are now referencing there.

```
==== Global mocks and parallel execution
Creating a global `GroovyMock`/`GroovyStub`/`GroovySpy` when <<parallel-execution.adoc#parallel-execution,parallel execution>> is enabled,
requires that the spec is annotated with <<parallel-execution.adoc#isolated-execution, @Isolated>> or `@ResourceLock(org.spockframework.runtime.model.parallel.Resources.META_CLASS_REGISTRY)`.
If it is only used for a feature, then it suffices that the feature is annotated with `@ResourceLock(org.spockframework.runtime.model.parallel.Resources.META_CLASS_REGISTRY)`.

```

**Comment by Vampire** [comment]
> not in another one referencing it

Well, you **are not** referencing it.

You reference "Global Groovy Mocks for Static Methods" a subsection of "Groovy Mocks".

Then section you mention is "Global mocks and parallel execution" which is a subsection of "Mocking All Instances of a Type", which again is a subsection of "Groovy Mocks".

This subsection is not even shown in the navigation as only first 2 levels of headings are shown.

So it really is just documented for using Global Groovy mocks generally, but in the linked-to section about using global Groovy mocks for static methods.

I thought it was there before, but you indeed only added this note with the original static-mock-maker PR.

Maybe we should at least add some link from "Global Groovy Mocks for Static Methods" to the "Global mocks and parallel execution" section?

**Pull request merged**